labcontrol/db/process.py

```diff
@@ -3244,18 +3248,40 @@ def generate_sample_sheet(self):
     def generate_prep_information(self):
         """Generates prep information
 
+        Returns
+        -------
+        dict labcontrol.db.study.Study: str
+            a dict of the Study and the prep
+        """
+        if self.is_amplicon_assay:
+            return self._generate_amplicon_prep_information()
+        elif self.is_metagenomics_assay:
+            return self._generate_metagenomics_prep_information()
+
+        raise ValueError("Prep file generation is not implemented for this "
```

Comment:
Wondering if is possible to get to this line? Also, is it possible to add a test?

Reply:
For better or worse, it appears so. From what I've gathered, a SequencingProcess object is defined as either Amplicon or Metagenomic during initialization by using either the 1 or 2 integers, respectively. Supplying 3 doesn't cause an error, as evidenced by the test_generate_prep_information_error() method successfully matching the observed results to the expected string.

Reply:
FYI, we anticipate more kinds of assays in the future, so everything we write has to anticipate >2 possible assay types (as this code does :)